portfolio/src/main/java/com/google/sps/servlets/DataServlet.java

```diff
@@ -82,4 +93,21 @@ private String getParameter(HttpServletRequest request, String name, String defa
     }
     return value;
   }
+
+  /** Returns the integer entered by user on how many comments  */
+  private int getNumDisplayComments(HttpServletRequest request) {
+    // Get the input from the form on MaxComments and convert to integer to the
+    // toGet function.
+    String CommentNumString = request.getParameter("max-comments");
+    // Convert the input to an int.
+    int CommentNum;
+    try {
+      CommentNum = Integer.parseInt(CommentNumString);
+    } catch (NumberFormatException e) {
+      System.err.println("Could not convert to int: " + CommentNumString);
+      return -1;
+    }
+
+    return CommentNum;
```

Comment:
You can just return directly from inside the `try` block, which saves you a few lines.